dll/win32/kernel32/client/synch.c

```diff
@@ -446,13 +446,47 @@ CreateSemaphoreW(IN LPSECURITY_ATTRIBUTES lpSemaphoreAttributes  OPTIONAL,
                  IN LONG lMaximumCount,
                  IN LPCWSTR lpName  OPTIONAL)
 {
-    CreateNtObjectFromWin32Api(Semaphore, Semaphore, SEMAPHORE,
+    CreateNtObjectFromWin32Api(Semaphore, Semaphore, SEMAPHORE_ALL_ACCESS,
                                lpSemaphoreAttributes,
                                lpName,
                                lInitialCount,
                                lMaximumCount);
 }
 
+HANDLE
+WINAPI
+DECLSPEC_HOTPATCH
+CreateSemaphoreExW(IN LPSECURITY_ATTRIBUTES lpSemaphoreAttributes  OPTIONAL,
+                 IN LONG lInitialCount,
+                 IN LONG lMaximumCount,
+                 IN LPCWSTR lpName  OPTIONAL,
+                 IN DWORD dwFlags OPTIONAL,
+                 IN DWORD dwDesiredAccess OPTIONAL)
+{
+    CreateNtObjectFromWin32Api(Semaphore, Semaphore, dwDesiredAccess,
+                               lpSemaphoreAttributes,
+                               lpName,
+                               lInitialCount,
+                               lMaximumCount);
+}
+
+HANDLE
+WINAPI
+DECLSPEC_HOTPATCH
+CreateSemaphoreExA(IN LPSECURITY_ATTRIBUTES lpSemaphoreAttributes OPTIONAL,
+                 IN LONG lInitialCount,
+                 IN LONG lMaximumCount,
+                 IN LPCSTR lpName  OPTIONAL,
+                 IN DWORD dwFlags OPTIONAL,
+                 IN DWORD dwDesiredAccess OPTIONAL)
+ConvertAnsiToUnicodePrologue
+    if (!lpName)
+        return CreateSemaphoreExW(lpSemaphoreAttributes, lInitialCount, lMaximumCount, NULL, dwFlags, dwDesiredAccess);
```

Comment:
Please go to newlines for each parameter (and then put braces for the `if` block).